### DISCUSSION TITLE
Wyszukiwanie w Azure storage [tagi]

### DISCUSSION THREAD

**Discussion by szympulka** [AzureCommunityPL/group#20] [Pomoc] [answered]
Chcę odejść od tradycyjnego sql i użyć azure storage ze względu na niezbyt dużą ilość danych. Potrzebuję wprowadzić coś na wzór tagów, aby móc wyszukiwać informacje po słowach kluczowych. Dotychczas samo TableQuery i QueryComparisons mi wystarczało. Google nie nakierował mnie jakoś na rozwiązanie. Jak najlepiej można to ogarnąć? Przyszła mi też do głowy pomocnicza tabela z tagami, i powiązanie ich coś na wzór relacyjnych baz danych.

**Comment by kaluzaaa** [comment, marked as answer]
@szympulka jak zwykle do dokumentacji 😉 Struktury KV to inny sposób myślenia, zapomnij o relacjach albo zostań przy SQL najtańszym. Więc albo zgodne ze sztuką modelowanie modelowanie, albo index w innej usłudze (Azure Search). 

1. https://docs.microsoft.com/en-us/azure/storage/tables/table-storage-design-guidelines
1. https://docs.microsoft.com/en-us/azure/storage/tables/table-storage-design-patterns
1. https://docs.microsoft.com/en-us/azure/storage/tables/table-storage-design-modeling
1. https://docs.microsoft.com/en-us/azure/storage/tables/table-storage-design-for-query
1. https://docs.microsoft.com/en-us/azure/search/search-howto-indexing-azure-tables

**Reply by kamil-mrzyglod** [reply]
Ja powiem ze swojego doświadczenia - kiedyś postawiłem u klienta rozwiązanie oparte o Table Storage bo chciałem być "edgy" zamiast tradycyjnego SQLa. Po roku przerobiliśmy całe rozwiązanie na Azure SQL bo całość się kupy nie trzymała pomimo zastosowania patternów z dokumentacji. Nawet dla małej ilości danych poszedłbym w Azure SQL jeśli masz relacje w danych bo zwyczajnie KV, jak Łukasz wspomniał, to inny sposób myślenia.

**Reply by szympulka** [reply]
Hmm, w takim razie chyba wrócę do tego najtańszego sql skoro tak to wygląda. Co prawda aplikacja nie kliencka, ale nie chce potem wracać do niej
